### PR TITLE
Only try and archive Mac & Win buildInstaller artifacts

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Signed-off-by: andrew-m-leonard <andrew_m_leonard@uk.ibm.com>

### CODE REVIEW

**Comment by andrew-m-leonard** [comment]
Test builds:
 https://ci.adoptopenjdk.net/view/work%20in%20progress/job/andrew-jdk11u-linux-x64-openj9/1/
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/andrew-jdk11u-mac-x64-openj9/1/
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/andrew-jdk11u-windows-x64-openj9/1/

**Comment by andrew-m-leonard** [comment]
Test builds look good

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -325,12 +325,16 @@ class Build {
...
                 	default: return; break
                 }
-		try {
-                    context.sh 'for file in $(ls workspace/target/*.tar.gz workspace/target/*.pkg workspace/target/*.msi); do sha256sum "$file" > $file.sha256.txt ; done'
-                    writeMetadata(versionData)
-                    context.archiveArtifacts artifacts: "workspace/target/*"
-                } catch (e) {
-                    context.println("Failed to build ${buildConfig.TARGET_OS} installer ${e}")
+                // Archive the Mac and Windows pkg/msi
+                if (buildConfig.TARGET_OS == "mac" || buildConfig.TARGET_OS == "windows") {
```
Comment (sxa) [inline pipelines/build/common/openjdk_build_pipeline.groovy:329]: Can we add a comment explaining why linux isn't in there for people looking at this in the future?

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 2]
```diff
@@ -325,12 +325,16 @@ class Build {
...
+		    try {
+                        context.sh 'for file in $(ls workspace/target/*.tar.gz workspace/target/*.pkg workspace/target/*.msi); do sha256sum "$file" > $file.sha256.txt ; done'
+                        writeMetadata(versionData)
+                        context.archiveArtifacts artifacts: "workspace/target/*"
+                    } catch (e) {
+                        context.println("Failed to build ${buildConfig.TARGET_OS} installer ${e}")
+                        currentBuild.result = 'FAILURE'
```
Comment (M-Davies) [inline pipelines/build/common/openjdk_build_pipeline.groovy:336]: This line should probably be removed as this was what was causing us a lot of false failures this morning https://github.com/AdoptOpenJDK/openjdk-build/pull/1787

Comment (sxa) [reply pipelines/build/common/openjdk_build_pipeline.groovy:336]: Is that directly related to this issue Morgan? That sounds like a separate PR if it's required

Comment (M-Davies) [reply pipelines/build/common/openjdk_build_pipeline.groovy:337]: The PR is adding it back into the master

**Comment by sxa** [comment]
Yep agreed - current behaviour as coded in here is correct :-)
Initiated [JDK11](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk11-pipeline/962/) and [JDK15](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk15-pipeline/135/) pipelines in an attempt to see if they run through ok before the nightlies kick off

**Comment by sxa** [comment]
xLinux JDK15 running cleanly after these changes so i'm hopeful that things will look better tomorrow than they have for the last week.

**Pull request merged**